### ISSUE TITLE
Bug: V 19.0.2 INVOICE CLONE KEEPS THE CLONED BANK ACCOUNT NUMBER FOR PAYMENT AND IS IMPOSSIBLE TO CHANGE IT (CLOSED IN DOLIBARR FEW YEARS AGO)

### ISSUE THREAD

**Issue by PlanPc** [Dolibarr/dolibarr#30414]
# Bug
Cloning an old invoice, keeps an old bank account as payment method and is always showed even if the account is changed at the invoice payment properties or the payment method is changed to other methods. The cloned bank account was closed for many years before that invoice clone. Never before happend since Doli V 3.

## Environment
- **Version**: 19.0.2
- **OS**: Linux
- **Web server**: nginx/1.16.1
- **PHP**: fpm-fcgi 7.4.30
- **Database**:  5.5.5-10.3.29-MariaDB
- **URL(s)**: /compta/facture/prelevement.php?id=XXXXX

## Expected and actual behavior
Since the old bank account of the original invoiced is closed, when cloning that invoice with bank as payment method, the bank account must be blank or at least must anyone with priviledges must be able to change and assing an active bank account at dolibarr as payment method to show in PDFs. Even activating the closed old bank account, try to change it at the invoice payment method, but always stil the same closed old bank accout, so is not possible to select an active bank account fot that cloned invoice.

## Steps to reproduce the behavior
Just cloning an old invoice, where the bank account related for payment is closed, theres is no way to change bank account at the invoice tab /compta/facture/prelevement.php?id=XXXXX.

## Report
Only changing manually the field in DB --> llx_facture --> fk_account is possible to update for a valid bank account

**Comment by sonikf** [comment]
Hi @PlanPc 
It works for me can't replicate!

**Comment by PlanPc** [comment]
Thank you @sonikf 

I recently update from 17.0.3 to 18.0.4 and then to 19.0.2, and just after is when this happen. This is an instance that comes from an early V 3.2.2 of dolibarr, regularly updated from one major/minor version to another. I wonder if that could be the reason for it to happen.

I have looked for formulas to try to clean up this installation in a way that allows configuration and data to be maintained, to be sure that there is no problem "carried over" for a long time, but to date I have not found a convincing method to do it.

If this could be a real bug I hopen can help developers to find and debug it, if it is not, this issue can be closed with no drama, since the casuistry must be very scarce and the solution is also not dramatic.

Regards

**Comment by sonikf** [comment]
> This is an instance that comes from an early V 3.2.2 of dolibarr, regularly updated from one major/minor version to another.

Wow! kudos to you!

Can you post screenshots of bug?
Have you tried modifying the original invoice temporarily before cloning as a workaround?


> I have looked for formulas to try to clean up this installation in a way that allows configuration and data to be maintained, to be sure that there is no problem "carried over" for a long time, but to date I have not found a convincing method to do it.

The only reliable method i can think is for you to back up, initiate a dev installation and check structure of every table in your db against the tables listed here https://github.com/Dolibarr/dolibarr/tree/19.0/htdocs/install/mysql/tables. Time consuming for sure but the only way to know what's missing!

**Comment by PlanPc** [comment]
> The only reliable method i can think is for you to back up, initiate a dev installation and check structure of every table in your db against the tables listed here https://github.com/Dolibarr/dolibarr/tree/19.0/htdocs/install/mysql/tables. Time consuming for sure but the only way to know what's missing!

Thank you for that info @sonikf 

I cant post any bug, but now I know that it won't be necessary either...

I just cloned exactly the same invoice that gave me that problem yesterday to record a short video that I could share here and, oh surprise, it worked perfectly now.

I should have done more testing before creating this issue, but I was focused on the fact that I updated to V 19.0.2 just before and assumed that it was some problem in the new version. You learn from everything, at every moment. Thank you very much and greetings
